addons/dragging/functions/fnc_startDrag.sqf

```diff
@@ -36,6 +36,16 @@ if !(GVAR(dragAndFire)) then {
     } else {
         _unit selectWeapon primaryWeapon _unit;
     };
+} else { // Making sure the unit is holding a primary weapon or handgun
+    private _allowedWeapons = [primaryWeapon _unit, handgunWeapon _unit];
+
+    if (_allowedWeapons findIf {_x isEqualTo (currentWeapon _unit)} == -1) then {
+        if ((_allowedWeapons select 0) isEqualto "") then {
+            _unit selectWeapon (_allowedWeapons select 1);
+        } else {
+            _unit selectWeapon (_allowedWeapons select 0);
+        };
+    };
```

Comment:
> Currently lacks handling for no weapons besides launcher.

Yeah, I've just tested it and having binoculars equipped when you start dragging will break it as well. Essentially, it plays the animation properly but the weapon info box displays the launcher or binocular as current weapon therefore you can't fire.

Maybe something like this could work:

```sqf
if (currentWeapon _unit == secondaryWeapon _unit || currentWeapon player isKindOf ["Binocular", configFile >> "CfgWeapons"]) then {
    if (primaryWeapon _unit != "") then {
        // Use primary if possible
        _unit selectWeapon primaryWeapon _unit;
    } else {
        if (handgunWeapon _unit != "") then {
            // Use pistol if unit has no primary
            _unit selectWeapon handgunWeapon _unit;
        } else {
            // Add fake weapon if no weapons besides launcher are available
            _unit addWeapon "ACE_FakePrimaryWeapon";
            _unit selectWeapon primaryWeapon _unit;
        };
    };
};
```

Is there any other case other than holding a launcher or binos? 🤔

Reply:
Ah right, forgot about binos. Updated my suggestion.